File: flutter/lib/data/session.dart
import 'dart:async';
import 'package:flutter/material.dart';
import 'package:shared_preferences/shared_preferences.dart';

class Session {
  String id_user = "id_user";
  String nama_user = "nama_user";
  String email_user = "email_user";
  String api_token = "api_token";
  String is_login = "is_login";

  //SETTER
  Future<bool> setId(int value) async {
    final SharedPreferences session = await SharedPreferences.getInstance();

    return session.setInt(id_user, value);
  }

  Future<bool> setNama(String value) async {
    final SharedPreferences session = await SharedPreferences.getInstance();

    return session.setString(nama_user, value);
  }

  Future<bool> email(String value) async {
    final SharedPreferences session = await SharedPreferences.getInstance();

    return session.setString(email_user, value);
  }

  Future<bool> token(String value) async {
    final SharedPreferences session = await SharedPreferences.getInstance();

    return session.setString(api_token, value);
  }

  Future<bool> setIsLogin(bool login) async {
    final SharedPreferences session = await SharedPreferences.getInstance();

    return session.setBool(is_login, login);
  }

  //GETTER
  Future<int> getId() async {
    final SharedPreferences session = await SharedPreferences.getInstance();

    return session.getInt(id_user) ?? "";
  }

  Future<String> getNama() async {
    final SharedPreferences session = await SharedPreferences.getInstance();

    return session.getString(nama_user) ?? "";
  }

  Future<String> getEmail() async {
    final SharedPreferences session = await SharedPreferences.getInstance();

    return session.getString(email_user) ?? "";
  }

  Future<String> getToken() async {
    final SharedPreferences session = await SharedPreferences.getInstance();

    return session.getString(api_token) ?? "";
  }

  Future<bool> getIsLogin() async {
    final SharedPreferences session = await SharedPreferences.getInstance();

    return session.getBool(is_login) ?? false;
  }

  //REMOVER
  Future logout() async {
    final SharedPreferences session = await SharedPreferences.getInstance();

    session.remove(id_user);
    session.remove(nama_user);
    session.remove(email_user);
    session.remove(api_token);
    session.remove(is_login);
  }

  //Login
  Future login(int id, String nama, String email, String apiToken) async {
    final SharedPreferences session = await SharedPreferences.getInstance();

    session.setInt(id_user, id);
    session.setString(nama_user, nama);
    session.setString(email_user, email);
    session.setString(api_token, apiToken);
    session.setBool(is_login, true);
  }
}

File: android/app/src/main/java/com/abdulr/lestaribuah/Data/Offline/Session.java
package com.abdulr.lestaribuah.Data.Offline;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;

public class Session {
    private static final String appName = "Toko Buah";
    private static final int id = 0;
    private static final String nama = "";
    private static final String email = "";
    private static final String noHp = "";
    private static final String token = "";
    private static final String tokenFirebase = "";
    private static final Boolean isLogin = false;
    private SharedPreferences sp;
    private SharedPreferences.Editor spEditor;

    @SuppressLint("CommitPrefEdits")
    public Session(Context context) {
        sp = context.getSharedPreferences(appName, Context.MODE_PRIVATE);
        spEditor = sp.edit();
    }

    //SETTER
    public void setId(int value) {
        spEditor.putInt("id", value);
        spEditor.commit();
    }

    public void setNama(String value) {
        spEditor.putString("nama", value);
        spEditor.commit();
    }

    public void setEmail(String value) {
        spEditor.putString("email", value);
        spEditor.commit();
    }

    public void setNoHp(String value) {
        spEditor.putString("noHp", value);
        spEditor.commit();
    }

    public void setToken(String value) {
        spEditor.putString("token",value);
        spEditor.commit();
    }

    public void setTokenFirebase(String value) {
        spEditor.putString("tokenFirebase",value);
        spEditor.commit();
    }

    public void setIsLogin (Boolean value) {
        spEditor.putBoolean("isLogin", value);
        spEditor.commit();
    }

    public void login(int id, String nama, String email, String noHp, String token, String tokenFirebase) {
        spEditor.putInt("id", id);
        spEditor.putString("nama", nama);
        spEditor.putString("email", email);
        spEditor.putString("noHp", noHp);
        spEditor.putString("token", token);
        spEditor.putString("tokenFirebase", tokenFirebase);
        spEditor.putBoolean("isLogin", true);
        spEditor.commit();
    }
    //ENDSETTER


    //GETTER
    public int getId() {
        return sp.getInt("id", id);
    }

    public String getNama() {
        return sp.getString("nama",nama);
    }

    public String getEmail() {
        return sp.getString("email",email);
    }

    public String getNoHp() {
        return sp.getString("noHp",noHp);
    }

    public String getToken() {
        return sp.getString("token",token);
    }

    public String getTokenFirebase() {
        return sp.getString("tokenFirebase",tokenFirebase);
    }

    public Boolean getIsLogin() {
        return sp.getBoolean("isLogin", isLogin);
    }

    public void logout() {
        spEditor.putInt("id", id);
        spEditor.putString("nama", nama);
        spEditor.putString("email", email);
        spEditor.putString("noHp", noHp);
        spEditor.putString("token", token);
        spEditor.putString("tokenFirebase", tokenFirebase);
        spEditor.putBoolean("isLogin", isLogin);
        spEditor.commit();
    }
    //ENDGETTER
}
